### PR TITLE
Percent refactor

### PR DESCRIPTION
**Pull request by Paul-Knapp**
percents now use a written private helper. all tests pass with 100% coverage

### CODE REVIEW

**Review by wally-yawn — APPROVED** [review]
These all look good to me

File: lib/stat_tracker.rb [hunk 1]
```diff
@@ -39,27 +39,24 @@ def self.from_csv(source)
   end
 
   def percentage_home_wins
-    total_games = @all_games.length
     home_wins = @all_games.count {|game| game.home_goals > game.away_goals}
           
-    percentage = (home_wins.to_f / total_games) * 100
-    percentage.round(2)
+    percentage = (home_wins.to_f / total_games)
+    percentage.round(4)
```
Comment (Crosswolfv1) [inline lib/stat_tracker.rb]: Should be rounding to 2

File: lib/stat_tracker.rb [hunk 2]
```diff
@@ -39,27 +39,24 @@ def self.from_csv(source)
...
   end
 
   def percentage_visitor_wins
-    total_games = @all_games.length
     visitor_wins = @all_games.count {|game| game.away_goals > game.home_goals}
           
-    percentage = (visitor_wins.to_f / total_games) * 100
-    percentage.round(2)
+    percentage = (visitor_wins.to_f / total_games) 
+    percentage.round(4)
```
Comment (Crosswolfv1) [inline lib/stat_tracker.rb]: Should be rounding to 2

File: lib/stat_tracker.rb [hunk 3]
```diff
@@ -39,27 +39,24 @@ def self.from_csv(source)
...
   end
 
   def percentage_ties
-    total_games = @all_games.length
     ties = @all_games.count {|game| game.away_goals == game.home_goals}
           
-    percentage = (ties.to_f / total_games) * 100
-    percentage.round(2)
+    percentage = (ties.to_f / total_games)
+    percentage.round(4)
```
Comment (Crosswolfv1) [inline lib/stat_tracker.rb]: Should be rounding to 2

File: spec/stat_tracker_spec.rb [hunk 4]
```diff
@@ -36,20 +36,20 @@
   end
   describe '#calculate percentages' do
     it 'can calculate home wins' do
-      expect(@stat_tracker1.percentage_home_wins).to eq(68.75)
+      expect(@stat_tracker1.percentage_home_wins).to eq(0.6875)
```
Comment (Crosswolfv1) [inline spec/stat_tracker_spec.rb]: Should be rounding to 2

File: spec/stat_tracker_spec.rb [hunk 5]
```diff
@@ -36,20 +36,20 @@
...
     end
 
     it 'can calculate visitor wins' do
-      expect(@stat_tracker1.percentage_visitor_wins).to eq(28.13)
+      expect(@stat_tracker1.percentage_visitor_wins).to eq(0.2813)
```
Comment (Crosswolfv1) [inline spec/stat_tracker_spec.rb]: Should be rounding to 2

File: spec/stat_tracker_spec.rb [hunk 6]
```diff
@@ -36,20 +36,20 @@
...
     end
 
     it 'can calculate ties' do
-      expect(@stat_tracker1.percentage_ties).to eq(3.13)
+      expect(@stat_tracker1.percentage_ties).to eq(0.0313)
```
Comment (Crosswolfv1) [inline spec/stat_tracker_spec.rb]: Should be rounding to 2

File: spec/stat_tracker_spec.rb [hunk 7]
```diff
@@ -36,20 +36,20 @@
...
     end
 
     it 'can calculate accurately' do
       total = (@stat_tracker1.percentage_ties) + (@stat_tracker1.percentage_visitor_wins) + (@stat_tracker1.percentage_home_wins)
-      expect(total).to be_within(0.03).of(100.00)
+      expect(total).to be_within(0.003).of(1.0000)
```
Comment (Crosswolfv1) [inline spec/stat_tracker_spec.rb:62]: Should be rounding to 2

**Review by Crosswolfv1 — CHANGES_REQUESTED** [review]
Should be rounding to 2 marked lines with comments

**Review by wally-yawn — APPROVED** [review]

File: lib/stat_tracker.rb (hunk 1 above)
Comment (Paul-Knapp) [reply lib/stat_tracker.rb]: should be fixed now thanks.

File: lib/stat_tracker.rb (hunk 2 above)
Comment (Paul-Knapp) [reply lib/stat_tracker.rb]: Should be fixed now thanks.

File: spec/stat_tracker_spec.rb (hunk 4 above)
Comment (Paul-Knapp) [reply spec/stat_tracker_spec.rb]: should be fixed now passes all tests.

File: spec/stat_tracker_spec.rb (hunk 6 above)
Comment (Paul-Knapp) [reply spec/stat_tracker_spec.rb]: tests are fixed now.

File: spec/stat_tracker_spec.rb (hunk 7 above)
Comment (Paul-Knapp) [reply spec/stat_tracker_spec.rb:62]: I have fixed it now.

**Review by Montana-Pfeifer — COMMENTED** [review]
Looked at concerns of other reviews and looks like all changes have been made to be able to merge

**Review by Crosswolfv1 — APPROVED** [review]
Requested changes have been made

**Pull request merged**